server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -31,6 +31,10 @@ package object server {
     val SocketAddress = InetSocketAddress.createUnresolved(Host, HttpPort)
   }
 
+  object ServerRequestKeys {
+    val SecureSession: AttributeKey[Option[SecureSession]] = AttributeKey[Option[SecureSession]]
+  }
+
```

Comment:
This is going to need to be reconciled with #2318, which is ready to go once we come to a conclusion on that Vault dependency.  Someone is going to have a light merge to do here.